src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -6,227 +6,199 @@
-        Success(cbf().result())
-      } else {
-        val tryBuilder = value.split(",").foldLeft(Try(cbf())) {
-          case (tryResult, rawValue) =>
-            for {
-              result <- tryResult
-              value <- stringConvert.value.from(rawValue)
-            } yield result += value
-        }
-
-        tryBuilder.map(_.result())
-      }
+    override def to(t: Option[T]): ConfigValue = t match {
+      case Some(v) => conv.value.to(v)
+      case None => ConfigValueFactory.fromMap(Map().asJava)
     }
 
-    override def to(ts: F[T], namespace: String): RawConfig =
-      Map(namespace -> ts.map(stringConvert.value.to).mkString(","))
+    def toOption(t: Option[T]): Option[ConfigValue] = t.map(conv.value.to)
   }
 
-  // traversable of complex types, that are types with an instance of ConfigConvert
-  implicit def deriveTraversableOfObjects[T, F[T] <: TraversableOnce[T]](implicit configConvert: Lazy[ConfigConvert[T]],
+  // traversable of types with an instance of ConfigConvert
+  implicit def deriveTraversable[T, F[T] <: TraversableOnce[T]](implicit configConvert: Lazy[ConfigConvert[T]],
     cbf: CanBuildFrom[F[T], T, F[T]]) = new ConfigConvert[F[T]] {
 
-    override def from(config: RawConfig, namespace: String): Try[F[T]] = {
-      val fullKeysFound = config.keys.filter(_ startsWith (namespace + namespaceSep)).toList
-      val keysFound = fullKeysFound.map { key =>
-        val parentNamespaceLength = namespace.length + namespaceSep.length
-        key.substring(0, key.indexOfSlice(namespaceSep, parentNamespaceLength))
-      }.sorted.distinct
-
-      if (keysFound.isEmpty) {
-        Success(cbf().result())
-      } else {
-        val tryBuilder = keysFound.foldLeft(Try(cbf())) {
-          case (tryResult, key) =>
-            for {
-              result <- tryResult
-              value <- configConvert.value.from(config, key)
-            } yield result += value
-        }
-
-        tryBuilder.map(_.result())
+    override def from(config: ConfigValue): Try[F[T]] = {
+      config match {
+        case co: ConfigList =>
+          val tryBuilder = co.asScala.foldLeft(Try(cbf())) {
+            case (tryResult, v) =>
+              for {
+                result <- tryResult
+                value <- configConvert.value.from(v)
+              } yield result += value
+          }
+
+          tryBuilder.map(_.result())
+        case _ =>
+          Success(cbf().result())
       }
     }
 
-    override def to(ts: F[T], namespace: String): RawConfig = {
-      val tsWithIndexes = ts.toList.zipWithIndex // give an index/id to each element of the traversable
-      val tsConverted = tsWithIndexes.map { case (t, i) => configConvert.value.to(t, makeNamespace(namespace, i.toString)) }
-      tsConverted.foldLeft(Map.empty[String, String])(_ ++ _)
+    override def to(ts: F[T]): ConfigValue = {
+      val tsList = ts.toList // give an index/id to each element of the traversable
```

Comment:
No need for the comment here and you can merge this expression with the one below. You can also remove the `toIterable` because `List` is `Iterable`, so this code can be rewritten as

```scala
ConfigValueFactory.fromIterable(ts.toList.map(configConvert.value.to).asJava)
```